test-integration/graql/reasoner/query/GenerativeOperationalIT.java

```diff
@@ -70,21 +71,38 @@
     public static final GraknTestStorage storage = new GraknTestStorage();
 
     private static Session genericSchemaSession;
+    private static HashMultimap<Pattern, Pattern> relationPatternTree;
 
     @BeforeClass
     public static void loadContext() {
         Config mockServerConfig = storage.createCompatibleServerConfig();
         genericSchemaSession = SessionUtil.serverlessSessionWithNewKeyspace(mockServerConfig);
         String resourcePath = "test-integration/graql/reasoner/resources/";
         loadFromFileAndCommit(resourcePath, "genericSchema.gql", genericSchemaSession);
+
+        try(Transaction tx = genericSchemaSession.readTransaction()) {
+            String id = tx.getEntityType("baseRoleEntity").instances().iterator().next().id().getValue();
+            String subId = tx.getEntityType("subRoleEntity").instances().iterator().next().id().getValue();
+            Pattern basePattern = and(
+                    var("r")
+                            .rel("subRole1", var("x"))
+                            .rel("subRole2", var("y"))
+                            .isa("binary"),
+                    var("x").isa("subRoleEntity"),
+                    var("x").id(id),
+                    var("y").isa("subRoleEntity"),
+                    var("y").id(subId)
+            );
+            relationPatternTree = generatePatternTree(basePattern, new TransactionContext(tx));
+        }
     }
 
     @AfterClass
     public static void closeSession() {
         genericSchemaSession.close();
     }
 
-    private Stream<Pair<Pattern, Pattern>> generateTestPairs(Pattern basePattern, TransactionContext ctx, boolean exhaustive){
+    private static HashMultimap<Pattern, Pattern> generatePatternTree(Pattern basePattern, TransactionContext ctx){
```

Comment:
what does this method do, exhaustively generate all patterns that are generalisations of the given pattern? Comment on this would be nice :)